lib/ansible/modules/windows/win_service.ps1

```diff
@@ -59,6 +61,65 @@ if ($null -ne $path) {
     $path = [System.Environment]::ExpandEnvironmentVariables($path)
 }
 
+Function ReversedHexValue ($v) {
+
+  $res = (("{0:x8}" -f $v) -split '(..)' | Where-Object { $_ }).split(" ")
+
+  [array]::Reverse($res)
+
+  $res = [system.string]::Join('', $res)
+
+  return $res
+}
+
+Function DecValue ($v) {
+
+  $res = ($v -split '(..)' | Where-Object { $_ }).split(" ")
+
+  [array]::Reverse($res)
+
+  $res = [system.string]::Join('', $res)
+
+  $res = [convert]::toint32($res, 16)
+
+  return $res
+}
+
+Function RecoveryActionMapping ($a) {
+
+    $res = switch($a) {
+        'no_action' { 0 }
+        'restart' { 1 }
+        'reboot' { 2 }
+        0 { 'no_action' }
+        1 { 'restart' }
+        2 { 'reboot' }
+        default { 0 }
+    }
+
+    return $res
+}
+
+Function Convert-RecoveryActionsToHuman ($s) {
+    $res = (("{0:x8}" -f $s) -split '(........)' | Where-Object { $_ }).split(" ")
+
+    for ($i=0; $i -lt $res.Length; $i++) {
+        $res[$i]=DecValue($res[$i])
+    }
+
+    $recovery_actions = @{
```

Comment:
We should be returning a list of recovery actions not just 1 dict. If there are no recovery actions then returning an empty list is probably ideal.